NoName-ForTeam/ui-lib
# Refactor/lib 59

# Configure Build Process and TypeScript Settings

## Changes

This PR introduces several improvements to our build process and TypeScript configuration:

- Added rimraf for cleaning the `dist` folder before each build, ensuring a fresh start for every build process.
- Updated `tsconfig.json` to explicitly set `outDir` and `rootDir`, providing clearer structure for our TypeScript compilation.
- Excluded Storybook stories and asset files from TypeScript compilation to optimize the build process.
- Updated project dependencies to ensure we're using the latest compatible versions.
- Updated `vite.config.ts` to align with our new build process requirements.

## Motivation

These changes aim to:
1. Streamline our build process
2. Improve TypeScript configuration clarity
3. Optimize compilation by excluding unnecessary files
4. Ensure we're using up-to-date dependencies

## Testing

To test these changes:
1. Clean install dependencies: `pnpm install`
2. Run the build process: `pnpm build`
3. Verify that the `dist` folder is properly generated without any TypeScript errors
4. Check that Storybook still works correctly: `pnpm storybook`

## Additional Notes

- Make sure to update any CI/CD pipelines if necessary to accommodate these changes.
- Review the updated `vite.config.ts` to ensure it meets all project requirements.

![Screenshot_8](https://github.com/user-attachments/assets/ce0885fe-8335-4185-8891-9a238e666dc0)
![Screenshot_1](https://github.com/user-attachments/assets/2a61ad8c-2929-4ebf-a702-382317ac5563)
